test-suite/buildtest.sh

```diff
@@ -16,6 +16,18 @@
 action="${1}"
 config="${2}"
 
+# Allow a layer to enable optional default-disabled features when
+# those features are supported in the current build environment
+# (and we can easily detect such support).
+if ${PKG_CONFIG:-pkg-config} --exists 'libecap >= 1.0 libecap < 1.1' 2>/dev/null
+then
+    CONFIGURE_FLAGS_MAYBE_ENABLE_ECAP="--enable-ecap"
```

Comment:
Please do not **force-enable** any feature. This particular will break both default and minimal build testing when libecap is installed in the OS for purposes of default build.

How this optional feature is supposed to be done is that **each** such optional feature has its own "layer-NN-FOO.opts" file(s) defining its build for each permutation of interaction with other options (eg --enable-ecap and --enable-adaptation, --disable-adaptation). Only elide such files if the permutation of build options is already existing in another layer (ie. minimal does the "--disable-adaptation --disable-ecap" case).

IIRC `buildtest.sh` is currently missing support for a layer `.opts` file to forcibly skip a test layer that cannot be run. That should be added instead of this.